### PR TITLE
Athena-58: License management changes

### LINKED ISSUE #58: Add return address and standard return text into license request notification.
The notification is currently sent from Athena [pallas@ohdsi.org]. Instead, I need it to go to the requesting user. So, please add as the return_to address the name and email address of the requesting user, e.g. "Zhang hui (1440859284@qq.com)". I am receiving quite a few of these, and have to copy and paste the email address from the text. Would make life easier.

### PR DESCRIPTION
**Pull request by acumarav**
Changes in the email templates, sending notifications, user's organization support 

Closes #58

### CODE REVIEW

File: properties/prod/application.properties [hunk 1]
```diff
@@ -100,6 +100,7 @@ athena.security.saml.attributes.email=email
 athena.security.saml.attributes.first_name=firstName
 athena.security.saml.attributes.last_name=lastName
 athena.security.saml.attributes.middle_name=middleName
+athena.security.saml.attributes.organization=organization
```
Comment (YaroslavTir) [inline properties/prod/application.properties]: It looks like there is a lot of duplication in properties. Do we have a base(common) `application.properties` that applies for all profiles? if so can you move all duplications there?

File: src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java [hunk 2]
```diff
@@ -292,20 +293,36 @@ public void checkBundleAndSharedUser(AthenaUser user, DownloadBundle bundle){
...
+                .collect(toList());
     }
 
     @Override
     public Long requestLicenses(AthenaUser user, Integer vocabularyV4Id) {
 
-        return saveLicenses(user, Collections.singletonList(vocabularyV4Id), PENDING).get(0).getId();
+        final List<License> requestedLicenses = buildLicenses(user, singletonList(vocabularyV4Id), PENDING);
+        requestedLicenses.forEach(license -> license.setRequestDate(new Date()));
+        return saveLicenses(user, requestedLicenses).get(0).getId();
```
Comment (YaroslavTir) [inline src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java]: this looks weird.  I would refactor this way: 
```
    @Override
    public Long requestLicense(AthenaUser user, Integer vocabularyV4Id) {

        final License license = buildLicense(user, vocabularyV4Id, PENDING);
        license.setRequestDate(new Date());

        final License savedLicense = licenseRepository.save(license);
        conceptService.invalidateGraphCache(user.getId());

        return savedLicense.getId();
    }

```

```
    private List<License> buildLicenses(AthenaUser user, List<Integer> vocabularyV4Ids, LicenseStatus status) {
        return vocabularyV4Ids.stream()
                .map(id -> buildLicense(user, id, status))
                .collect(toList());
    }

    private License buildLicense(AthenaUser user, Integer vocabularyV4Id, LicenseStatus status) {
        VocabularyConversion vocabularyConversion = new VocabularyConversion();
        return new License(user, vocabularyConversion, status)
    }
```

File: src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java [hunk 3]
```diff
@@ -292,20 +293,36 @@ public void checkBundleAndSharedUser(AthenaUser user, DownloadBundle bundle){
...
+    @Override
+    public List<License> saveLicenses(AthenaUser user, List<License> licenses) {
+
+        final List<License> savedLicenses = licenseRepository.save(licenses);
+
+        conceptService.invalidateGraphCache(user.getId());
+        return savedLicenses;
+    }
+
+    private List<License> buildLicenses(AthenaUser user, List<Integer> vocabularyV4Ids, LicenseStatus status) {
```
Comment (YaroslavTir) [inline src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java]: `private` methods should be at the end of the class

File: src/main/java/com/odysseusinc/athena/service/mail/EmailSenderService.java [hunk 4]
```diff
@@ -35,50 +36,62 @@
...
+    public void send(EmailType messageType, Map<String, Object> parameters, String... toEmails) {
 
-        if (emails == null || emails.length == 0 || StringUtils.isBlank(emails[0])) {
+        this.send(messageType, parameters, Arrays.asList(toEmails), emptyList());
+    }
+
+
+    public void send(EmailType messageType, Map<String, Object> parameters, List<String> toEmails, List<String> bccEmails) {
+
+        if (CollectionUtils.isEmpty(toEmails) && CollectionUtils.isEmpty(bccEmails)) {
```
Comment (YaroslavTir) [inline src/main/java/com/odysseusinc/athena/service/mail/EmailSenderService.java]: I can see from the code that `toEmails` can be empty if `bccEmails` has values. Is this right? bcc without main-emails looks strange.

File: src/main/resources/db/migration/athena/V20200310184726__schema_users_organization.sql [hunk 5]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE users
+    ADD COLUMN IF NOT EXISTS organization VARCHAR(100);
```
Comment (YaroslavTir) [inline src/main/resources/db/migration/athena/V20200310184726__schema_users_organization.sql]: I would set 255 just in case

Comment (acumarav) [reply src/main/resources/db/migration/athena/V20200310184726__schema_users_organization.sql]: done

File: src/main/java/com/odysseusinc/athena/service/mail/EmailSenderService.java (hunk 4 above)
Comment (acumarav) [reply src/main/java/com/odysseusinc/athena/service/mail/EmailSenderService.java]: I do not see a problem here

File: pom.xml [hunk 6]
```diff
@@ -115,33 +113,27 @@
...
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
-            <version>${spring.boot.version}</version>
         </dependency>
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-tomcat</artifactId>
-            <version>${spring.boot.version}</version>
             <!--<scope>provided</scope>-->
```
[the remark this answers is not in the document]
Comment (acumarav) [reply pom.xml]: done

File: pom.xml [hunk 7]
```diff
@@ -280,7 +272,11 @@
             <scope>system</scope>
             <systemPath>${project.basedir}/src/main/resources/opencsv.jar</systemPath>
         </dependency>
-
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>28.2-jre</version>
```
[the remark this answers is not in the document]
Comment (acumarav) [reply pom.xml]: done

File: pom.xml [hunk 8]
```diff
@@ -65,7 +64,6 @@
                             <!--<dependency>-->
                                 <!--<groupId>org.springframework.boot</groupId>-->
                                 <!--<artifactId>spring-boot-starter-tomcat</artifactId>-->
-                                <!--<version>${spring.boot.version}</version>-->
                             <!--</dependency>-->
                         <!--</dependencies>-->
```
[the remark this answers is not in the document]
Comment (acumarav) [reply pom.xml]: done

File: src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java (hunk 3 above)
Comment (acumarav) [reply src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java]: ok

File: properties/prod/application.properties (hunk 1 above)
Comment (acumarav) [reply properties/prod/application.properties]: no, there is not base properties

File: src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java (hunk 2 above)
Comment (acumarav) [reply src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java]: I've made it shorter and cleaner instead.

**Pull request merged**